### DISCUSSION TITLE
Unpublish non-compatible packages

### DISCUSSION THREAD

**Discussion by AlexWayfer** [pulsar-edit/.github#124] [Community Packages]
Hello.

Great work with the packages repository cloning and managing.

Although I've faced at least 1 package non-compatible with the latest Atom versions, therefore Pulsar's any version.

https://web.pulsar-edit.dev/packages/term3

![image](https://user-images.githubusercontent.com/6510020/216344220-044cb67e-b015-4cf7-b52a-49cf06dd3dc1.png)

![image](https://user-images.githubusercontent.com/6510020/216344330-0fb3a898-f628-49d9-82dc-a6be8f85d3b5.png)

Not sure is it a right place to inform about something like this.

**Comment by AlexWayfer** [comment]
Oh, I found https://github.com/orgs/pulsar-edit/discussions/26, and I can't delete this discussion now, sorry.

**Reply by Daeraxa** [reply]
That one is about archiving repos that were "inherited" from Atom but we might not want to continue, not necessarily Atom packages.

**Comment by Daeraxa** [comment]
So I think the best thing is that we will probably continue to collect data about packages like this and add flags to them to allow people to make a more informed decision as to what they are downloading.

So for something like this where it is clear that it is broken, has been broken for a long time and doesn't appear like a fix is forthcoming, I would advocate that we add a flag to it to make it obvious.

We also plan to add a tag to make actively developed packages clear where they are published manually to the repository. I'm not quite sure where we are up to with either of these discussions though.

**Reply by confused-Techie** [reply]
Yeah this is probably the perfect use of the badges and flags we want to enable on packages. To say this is known to be broken, but I am a fan of keeping it simply because then someone can find a basis for the code they want, could fork the existing package, make the changes needed to get it working then publish a newer version.

Then when they do they could always tell us and we can add an additional flag to point to the newer version.

Which with all that said, finding broken packages like @AlexWayfer has helpfully done here is good to know so that we can properly flag broken packages once these features exist.

**Reply by AlexWayfer** [reply]
I'm not sure about need for "broken" tag. I think there can be just version locks, and I think I already saw something like that in Atom Packages. Probably they should be managed also from the repository side, tuned.

**Reply by Daeraxa** [reply]
Not sure how or why we would want to base it on versions. We don't plan to support any Pulsar version except the most up to date one(s) (rolling vs regular release discussion aside). The packages themselves can version lock, sure, but if a package hasn't been updated in 7 years but still works and is unmaintained then it isn't going to get a Pulsar specific version update any time soon.

**Reply by confused-Techie** [reply]
@AlexWayfer Well "Broken" is probably a bit aggressive, and not the actual wording we would use. Maybe we could go more "Depreciated since Atom v1.60" or something like that. 

To get the same idea across, that essentially you shouldn't expect this package to work
